src/cargo/sources/git/utils.rs

```diff
@@ -265,13 +265,18 @@ impl<'a> GitCheckout<'a> {
     /// is done. Use [`GitCheckout::is_fresh`] to check.
     ///
     /// * The `database` is where this checkout is from.
-    /// * The `repo` will be the checked out Git repoistory at `path`.
+    /// * The `repo` will be the checked out Git repoistory.
     fn new(
-        path: &Path,
         database: &'a GitDatabase,
         revision: git2::Oid,
         repo: git2::Repository,
     ) -> GitCheckout<'a> {
+        let path = if repo.is_bare() {
+            repo.path()
+        } else {
+            repo.path().parent().unwrap()
```

Comment:
Do we want `.path().parent()` or [`.workdir()`](https://docs.rs/git2/latest/git2/struct.Repository.html#method.workdir)?